### PR TITLE
Feature/double shutdown 2

### PR DESCRIPTION
**Pull request by andre-merzky**
Mark, I hope you forgive the reduced diff readability, and appreciate the pep-piness :)

### CODE REVIEW

File: src/radical/pilot/controller/pilot_launcher_worker.py [hunk 1]
```diff
@@ -306,6 +306,19 @@ def run(self):
 
                 if self._disabled.is_set():
                     # don't process any new pilot start requests.  
+                    # NOTE: this is not clean, in principle there could be other
+                    #       launchers alive which want to still start those 
+                    #       pending pilots.  In practice we only ever use one
+                    #       pmgr though, and its during its shutdown that we get
+                    #       here...
```
Comment (marksantcroos) [inline src/radical/pilot/controller/pilot_launcher_worker.py]: To be clear, we are not ruling out the use of two pmgr's here are we?

Comment (andre-merzky) [inline src/radical/pilot/controller/pilot_launcher_worker.py]: No, I don't think so.  But in principle two launch controllers could be used for one pmgr, and this commit explicitly breaks that.  Well, OTOH, this is already broken in other places, so it doesn't change much :/

Comment (marksantcroos) [inline src/radical/pilot/controller/pilot_launcher_worker.py]: > ut in principle two launch controllers could be used for one pmgr, and this commit explicitly breaks that.

So as long as one doesn't explicitly instantiate the PM with multiple workers all is fine?

Comment (andre-merzky) [inline src/radical/pilot/controller/pilot_launcher_worker.py]: Yes

...

**Comment by marksantcroos** [comment]
Thanks! Looking forward to the refactoring though :-)

**Pull request merged**